packages/story-editor/src/components/floatingMenu/layer.js

```diff
@@ -92,16 +104,30 @@ function FloatingMenuLayer() {
     // Update now
     updatePosition();
 
-    // And update when certain elements' properties update
+    // And update when any element's properties inside the moveable box changes
     const observer = new MutationObserver(updatePosition);
-    MUTATE_SELECTORS.map((selector) => document.querySelector(selector))
-      .filter(Boolean)
-      .forEach((node) => observer.observe(node, { attributes: true }));
+    const node = document.querySelector('.moveable-control-box');
+    observer.observe(node, {
+      attributes: true,
+      subtree: true,
+      attributeFilter: ['style'],
+    });
```

Comment:
@merapi, I fixed the bug with the menu not registering some moveable changes by listening to attribute changes on the entire tree below the control-box element. It sounds expensive, but doesn't actually seem to be. There are up to 18 elements and for e.g. a rotation, all of them change their style attribute, but all the changes are reported in a single observe event with just a list of the 18 change records.

It seems to perform well, but please test it out and see if you can spot any problems. Thanks 🙏

Reply:
LG, I didn't find any issues.